### PR TITLE
add `delete` for key[s] in a NamedTuple

### PR DESCRIPTION
**Pull request by JeffreySarnoff**
Replaces #27725

This PR updates [delete for NamedTuples](https://github.com/JuliaLang/julia/pull/27725). See that (closed) PR for much discussion.  This revision supports deleting one or more keys from a NamedTuple.

### CODE REVIEW

File: base/namedtuple.jl [hunk 1]
```diff
@@ -404,11 +407,35 @@ end
...
+```jldoctest
+julia> delete((a=1, b=2, c=3), :a)
+(b = 2, c = 3)
+julia> delete((a=1, b=2, c=3), :b)
+(a = 1, c = 3)
+julia> delete((a=1, b=2, c=3), (:c, :b))
+(a = 1,)
+```
+"""
+function delete(a::NamedTuple{an}, @nospecialize(key::Symbol)) where {an}
```
Comment (Tokazama) [inline base/namedtuple.jl:431]: Why is there no specialize on `key::Symbol`?

File: base/namedtuple.jl [hunk 2]
```diff
@@ -404,11 +407,35 @@ end
...
+    names = diff_names(an, (key,))
+    NamedTuple{names}(a)
+end
+
+function delete(a::NamedTuple{an}, @nospecialize(keys::Tuple{Vararg{Symbol}})) where {an}
```
Comment (Tokazama) [inline base/namedtuple.jl:436]: I know it's a pain, but there should probably be a serious conversation concerning when tuples of indices are used and how they're interpreted. We have tuples of indices for a single index on multidim arrays, but how should we refer to multiple indices on tuples and named tuples? There's a loss of information if we require it always be another subtype of `AbstractArray`

File: base/exports.jl [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 export
-# Modules
+    # Modules
```
Comment (vchuravy) [inline base/exports.jl:4]: Please do not change formatting unecessarily

Comment (JeffreySarnoff) [reply base/exports.jl:4]: that was unintentional -- thank you for catching the mistype

File: base/namedtuple.jl (hunk 1 above)
Comment (JeffreySarnoff) [reply base/namedtuple.jl:431]: Thank you. I am removing it.

**Comment by JeffreySarnoff** [comment]
the test failures do not involve namedtuples.jl

**Comment by JeffreySarnoff** [comment]
This PR introduces `delete` to convey that the result is newly constructed (and the original is unaltered).
Perhaps `omit` a the better name (considering  [@Tokazama](https://github.com/JuliaLang/julia/pull/51098#discussion_r1309188308))?

 _experimental_ [`keep` and `omit` for NamedTuples](https://github.com/JeffreySarnoff/NamedTupleTools.jl/blob/using-indices/src/use_indices.jl). both accept (a) one key or index and (b) a Tuple of keys or indices.

**Comment by JeffBezanson** [comment]
This has many accidental formatting changes. No matter; the old version of the PR is just fine and we can merge that (it needs to be re-created though).

**Pull request closed without merging**